crates/runtime/src/dataconnector/mysql.rs

```diff
@@ -186,4 +191,205 @@ impl DataConnector for MySQL {
             }
         }
     }
+
+    fn metrics_provider(&self) -> Option<Arc<dyn MetricsProvider>> {
+        Some(Arc::new(MySQLMetricsProvider::new(
+            self.mysql_factory.conn_pool_metrics(),
+        )))
+    }
+}
+
+#[derive(Debug, Clone)]
+struct MySQLMetricsProvider {
+    metrics: Arc<Metrics>,
+}
+
+impl MySQLMetricsProvider {
+    fn new(metrics: Arc<Metrics>) -> Self {
+        Self { metrics }
+    }
+}
+
+const METRICS: &[MetricSpec] = &[
+    MetricSpec::new("connection_count", MetricType::ObservableGaugeU64)
```

Comment:
The actual metric definitions from the `mysql_async` crate. I'm only exposing the counters/gauges. We could in theory support the histograms as well - but that can come in a follow-up improvement.